Failing Assertion Error in FieldTypePluginManager due to missing field type categories configuration file

## Problem/Motivation
There's a failing Assertion in Drupals [FieldTypePluginManager](https://git.drupalcode.org/project/drupal/-/blob/10.3.11/core/lib/Drupal/Core/Field/FieldTypePluginManager.php?ref_type=tags#L195) class. Depending on how your debugging is configured this could lead to a fatal error "AssertionError: "AZ Quickstart" must be defined in MODULE_NAME.field_type_categories.yml in assert() (line 195 of core/lib/Drupal/Core/Field/FieldTypePluginManager.php).". 


### To Reproduce
Steps to reproduce the behavior:

<!--- Edit and replace examples below. Add or remove as needed  -->

1. Set `assert_options(ASSERT_ACTIVE, TRUE);` in Drupal settings or set the PHP config `assert.active` to on.
2. Try adding a new field to any of the content types.
3. You should receive the error " "AZ Quickstart" must be defined in MODULE_NAME.field_type_categories.yml in assert() " with a stack trace.

## Proposed resolution
Looks like we need a `az_quickstart.field_type_categories.yml` file to define the category specified in FieldType definitions in AZ Quickstart. Currently the definitions for `AZCardItem`  and `AZAccordionItem` set the category to `category: new TranslatableMarkup("AZ Quickstart")`, but "field type category as a translatable label in annotation is deprecated" ([reference](https://www.drupal.org/node/3375748)).
### Expected behavior

<!--- Edit and replace examples below. Add or remove as needed  -->

Adding a new field to Content Types should not result in an Assertion Error.

--- a/modules/custom/az_accordion/src/Plugin/Field/FieldType/AZAccordionItem.php
+++ b/modules/custom/az_accordion/src/Plugin/Field/FieldType/AZAccordionItem.php
@@ -21,7 +21,6 @@ use Drupal\Core\TypedData\DataDefinition;
 #[FieldType(
   id: "az_accordion",
   label: new TranslatableMarkup("Accordion"),
-  category: new TranslatableMarkup("AZ Quickstart"),
   default_widget: "az_accordion",
   default_formatter: "az_accordion_default",
 )]

--- a/modules/custom/az_card/src/Plugin/Field/FieldType/AZCardItem.php
+++ b/modules/custom/az_card/src/Plugin/Field/FieldType/AZCardItem.php
@@ -23,7 +23,6 @@ use Drupal\Core\TypedData\MapDataDefinition;
 #[FieldType(
   id: "az_card",
   label: new TranslatableMarkup("Card"),
-  category: new TranslatableMarkup("AZ Quickstart"),
   default_widget: "az_card",
   default_formatter: "az_card_default",
 )]

